R/README.Rmd

```diff
@@ -0,0 +1,132 @@
+---
+title: "RKDB DOCUMENTATION"
+author: 
+output:
+  md_document:
+    variant: markdown_github
+params:
+  server: localhost
+  port: 1234
+---
+
+```{r hooks, include=FALSE}
+# add stringr
+if(! 'gsubfn' %in% rownames(installed.packages())) install.packages('gsubfn')
+library(gsubfn)
+default_source_hook <- knitr::knit_hooks$get('source')
+knitr::knit_hooks$set(source = function(x, options) {
+  x <- gsubfn(x = x, pattern = "params\\$`?([\\w_]+)`?", function(y) {
+    y <- get(y, params)
+    ifelse(is.character(y), paste0("'", y, "'"), y)
+  })
+  default_source_hook(x, options)
+})
+```
+
+# Installation
+```{r install 1, echo=TRUE, message=FALSE, warning=FALSE}
+# remove old package
+if('qserver' %in% rownames(installed.packages())) remove.packages('qserver')
+# install devtools
+if(! 'devtools' %in% rownames(installed.packages())) install.packages('devtools')
+library(devtools)
+# install rkdb
+devtools::install_github('kxsystems/rkdb', quiet=TRUE)
+library(rkdb)
+```
+
+# First steps
+## Open a q server and connect to it
+Open a qserver to test the installation
+```{r start kdb, echo=TRUE, eval=FALSE}
+q -p params$port
+```
+
+Open a connection to it
+```{r open connection, echo=TRUE}
+h <- open_connection(params$server,params$port) #this open a connection
+```
+
+## Hello kdb
+```{r test 1, echo=TRUE}
+execute(h, '1+1')
+```
+
+Assigning a variable in q workspace:
+```{r test 2, echo=TRUE}
+execute(h, 'x:1+1') #assign x hopefully to 2
+execute(h, 'x') # get back the value
+```
+
+# Checks
+## Atoms from kdb to R
+
+As per [Q for mortals](http://code.kx.com/q4m3/2_Basic_Data_Types_Atoms/) kdb uses the following basic types:
+
+```{r atoms 0, echo=FALSE, warning=FALSE, message=FALSE}
+# install data.table
+if(! 'data.table' %in% rownames(installed.packages())) install.packages('data.table')
```

Comment:
no need to put a table here. Link to code.kx.com is enough. Table doesn't reflect mapping K<->R anyway